dashboard/src/components/shared/Dropdown.vue

```diff
@@ -15,7 +15,7 @@
 				<button class="button is-primary" type="button" slot="trigger">
 					<template v-if="selected">
 						<b-icon icon="users"></b-icon>
-						<span>{{selected}}</span>
+						<span>{{selected.slice(0,20)}}</span>
```

Comment:
```suggestion
						<span>{{selected | shortAddress )}}</span>
```

Try to make reusable Vue Filter (declare in `main.ts`)
```ts
Vue.filter('shortAddress', function (value: string) {
  return value ? value.slice(0,20) : '';
})
```